examples/domainmodel/src/cli/cli-util.ts

```diff
@@ -0,0 +1,48 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import fs from 'fs';
+import colors from 'colors';
+import { AstNode, DefaultDocumentValidator, LangiumDocument, LangiumDocumentConfiguration, LangiumServices } from 'langium';
+
+export function extractDocument(fileName: string, languageId: string, extensions: string[], services: LangiumServices): LangiumDocument {
+    if (!new RegExp(`^.*${extensions?.join('|')}$`).test(fileName)) {
+        console.error(`Please, choose a file with one of these extensions: ${extensions}.`);
+        process.exit(1);
+    }
+
+    if (!fs.existsSync(fileName)) {
+        console.error(`File ${fileName} doesn't exist.`);
+        process.exit(1);
+    }
+    const fileContent = fs.readFileSync(fileName, 'utf-8');
+
+    const document = LangiumDocumentConfiguration.create(`file:${fileName}`, languageId, 0, fileContent);
+    services.documents.DocumentBuilder.build(document);
+    if (!document.parseResult) {
+        console.error(`Failed to parse the grammar file ${fileName}`);
+        process.exit(1);
+    }
+
+    const validationErrors = new DefaultDocumentValidator(services)
+        .validateDocument(document)
```

Comment:
Validation is already done by the DocumentBuilder. Please use the BuildResult object, wich contains the parseResult (so there's no need to check whether it's present or not) and the diagnostics.